src/opm/parser/eclipse/EclipseState/Grid/EclipseGrid.cpp

```diff
@@ -331,18 +553,441 @@ namespace Opm {
+            coord.push_back(x0);
+            coord.push_back(yb);
+            coord.push_back(zb);
+
+            for (int i=0; i < dims[0]; i++) {
+
+                ind = i+(j+1)*dims[0]+1;
+
+                if (j == (dims[1]-1) ) {
+                    ind = i+j*dims[0]+1;
+                }
+
+                if (i == (dims[0] - 1) ) {
+                    ind=ind-1;
+                }
+
+                zt = tops[ind];
+                zb = zt + sumDZ(i, j, dims, dz);
+
+                double xt=-999;
+                double xb=-999;
+
+                if (j == (dims[1]-1) ) {
+                    xt = sumDX(j, 0, i, dims, dx);
+                    xb = sumDX(j, dims[2]-1, i, dims, dx);
+                } else {
+                    xt = sumDX(j+1, 0, i, dims, dx);
+                    xb = sumDX(j+1, dims[2]-1, i, dims, dx);
+                }
+
+                if (i == (dims[0] - 1) ) {
+                    yt = sumDY(i, 0, j, dims, dy);
+                    yb = sumDY(i, dims[2]-1, j, dims, dy);
+                } else {
+                    yt = sumDY(i+1, 0, j, dims, dy);
+                    yb = sumDY(i+1, dims[2]-1, j, dims, dy);
+                }
+
+                coord.push_back(xt);
+                coord.push_back(yt);
+                coord.push_back(zt);
+                coord.push_back(xb);
+                coord.push_back(yb);
+                coord.push_back(zb);
+
+                x0 = xt;
+            }
+        }
+
+        return coord;
+    };
+
+    std::vector<double> EclipseGrid::makeZcornDzTops(const std::array<int, 3>& dims, const std::vector<double>& dz, const std::vector<double>& tops) const {
+
+        std::vector<double> zcorn;
+        size_t sizeZcorn = dims[0]*dims[1]*dims[2]*8;
+
+        zcorn.assign (sizeZcorn, 0.0);
+
+        for (int j = 0; j < dims[1]; j++) {
+            for (int i = 0; i < dims[0]; i++) {
+                int ind = i + j*dims[0];
+                double z = tops[ind];
+
+                for (int k = 0; k < dims[2]; k++) {
+
+                    // top face of cell
+                    int zind = i*2 + j*dims[0]*4 + k*dims[0]*dims[1]*8;
+
+                    zcorn[zind] = z;
+                    zcorn[zind + 1] = z;
+
+                    zind = zind + dims[0]*2;
+
+                    zcorn[zind] = z;
+                    zcorn[zind + 1] = z;
+
+                    z = z + dz[i + j*dims[0] + k*dims[0]*dims[1]];
+
+                    // bottom face of cell
+                    zind = i*2 + j*dims[0]*4 + k*dims[0]*dims[1]*8 + dims[0]*dims[1]*4;
+
+                    zcorn[zind] = z;
+                    zcorn[zind + 1] = z;
+
+                    zind = zind + dims[0]*2;
+
+                    zcorn[zind] = z;
+                    zcorn[zind + 1] = z;
+                }
+            }
+        }
+
+        return zcorn;
+    };
+
+    double EclipseGrid::sumDX(int j, int k, int i1, const std::array<int, 3>& dims, const std::vector<double>& dx) const {
```

Comment:
This is in my opinion a `sumI` and not a `sumDX` function? Same with the `sumDY` and `sumDZ`.

Reply:
I disagree. This is actually a sum of DX used to calculate the X coordinate. It is a sum in I direction.

Reply:
> This is actually a sum of DX used to calculate the X coordinate. It is a sum in I direction.

Well - my point is that this a function which sums over the `I` direction; it is intended to be used on `DX` data - but there is nothing in the function which differentiates between `DX` and `DY/DZ` - i.e. this would "work" perfectly:
```
auto value = this->sumDX(j, k, i, DY);
```
as I see it you have named the function after the variable you intend to pass to it - and not after what it actually does - which is to sum over the `I` direction. That it is indeed used to sum over `DX` in the current code is determined at callsite - by passing the variable `DX` - and not in the in function itself.

I also really think the ordering of the arguments should be `(i,j,k)` - permuting that extremely well known ordering I think will confuse people.

Anyway - no further comments - your call.

Reply:
you are correct, I see this now and thanks for explaining this in more detail. 

changed from sumDX to sumIdir, sumDY to sumJdir and sumDZ to sumKdir